### PR TITLE
Replace some mapNotNull operators that can use new Kotlin 1.5 APIs.

### PR DESCRIPTION
**Pull request by vRallev**

### CODE REVIEW

File: compiler-utils/src/main/java/com/squareup/anvil/compiler/internal/PsiUtils.kt [hunk 1]
```diff
@@ -172,7 +172,7 @@ public inline fun <reified T> KtAnnotationEntry.findAnnotationArgument(
   // First check if the is any named parameter. Named parameters allow a different order of
   // arguments.
   annotationValues
-    .mapNotNull { valueArgument ->
+    .firstNotNullOfOrNull { valueArgument ->
```
Comment (0xMatthewGroves) [inline compiler-utils/src/main/java/com/squareup/anvil/compiler/internal/PsiUtils.kt:175]: Didn't realize I'd see this crazy method called in production code so soon 😄

**Review by 0xMatthewGroves — APPROVED** [review]

File: compiler-utils/src/main/java/com/squareup/anvil/compiler/internal/PsiUtils.kt (hunk 1 above)
Comment (lmchew) [reply compiler-utils/src/main/java/com/squareup/anvil/compiler/internal/PsiUtils.kt:175]: This name makes my eyes cross a little.

**Review by lmchew — APPROVED** [review]

File: compiler-utils/src/main/java/com/squareup/anvil/compiler/internal/PsiUtils.kt (hunk 1 above)
Comment (vRallev) [reply compiler-utils/src/main/java/com/squareup/anvil/compiler/internal/PsiUtils.kt:175]: The language lead indirectly replied to this here: https://twitter.com/relizarov/status/1390691109093920768?s=20

Comment (lmchew) [reply compiler-utils/src/main/java/com/squareup/anvil/compiler/internal/PsiUtils.kt:175]: True, naming is hard.

**Pull request merged**